dashboard/app/models/course.rb

```diff
@@ -189,21 +194,37 @@ def self.valid_courses
     end
   end
 
+  # @param user [User]
+  # @returns [Boolean] Whether the user has any experiment enabled which is
+  #   associated with an alternate course script.
+  def self.has_any_course_experiments?(user)
+    Experiment.any_enabled?(user: user, experiment_names: CourseScript.experiments)
+  end
+
+  # Get the set of valid courses for the dropdown in our sections table, using
+  # any alternate scripts based on any experiments the user belongs to.
+  def self.courses_for_user_with_experiments(user)
+    Course.
+      all.
+      select {|course| ScriptConstants.script_in_category?(:full_course, course[:name])}.
```

Comment:
Is this just to filter out PLC courses? I wonder if this should instead be something like `Course.where.not(:plc_course)`?

Reply:
For the most part, however it also appears to filter out `allthethingscourse.course`. More generally, what we have here is whitelisting, and what you're proposing would be blacklisting, so I think I'd rather stick with what's here.